ui/src/pages/AllStories/reducer.ts

```diff
@@ -0,0 +1,123 @@
+import { StoryView } from "../AllStories";
+
+export interface State {
+  tabValue: number;
+  visibleState: StoryView[];
+  visibleTableState: StoryView[];
+  tableData: StoryView[];
+  changedVisibility: StoryView[];
+  selectedRowIds: number[];
+  order: "asc" | "desc";
+  orderBy: string;
+}
+
+export type Action =
+  | { type: "SWITCH_TAB"; id: number }
+  | { type: "HANDLE_SWITCH_CHANGE"; e: React.ChangeEvent; story: StoryView }
```

Comment:
To write a test for the react change event we can just mock it out, and then cast it to make typescript happy.

```ts
const changeEvent = {
    target: {
      checked: true, // replace with false depending on what condition you wanna test
    },
  } as React.ChangeEvent;
```